src/Kinto.elm

```diff
@@ -63,6 +64,15 @@ type Endpoint
     | RecordEndpoint BucketName CollectionName RecordId
 
 
+type alias Body =
+    Encode.Value
+
+
+type Request
+    = KintoRequest Config Endpoint Method
+    | KintoRequestWithBody Config Endpoint Method Body
```

Comment:
Hmm. This allows some combinations of request parameters that I wouldn't have expected to be OK, such as `KintoRequestWithBody config BucketListEndpoint Get someBody` or `KintoRequest config BucketListEndpoint Delete`. But I guess it also permits certain things, such as POSTing a record with an ID (to a certain endpoint) vs. POSTing a record without an ID because you want one to be assigned by the server. I can't think of any better way to do this without having to spell out every possible combination of endpoint and method.